nautobot/dcim/management/commands/migrate_location_contacts.py

```diff
@@ -0,0 +1,205 @@
+                        break
+                    elif choice == "c":
+                        selected_contact = self.create_new_contact_from_location(location, model=Contact)
+                        break
+                    elif choice == "t":
+                        selected_contact = self.create_new_contact_from_location(location, model=Team)
+                        break
+                    elif choice.isdigit() and 0 < int(choice) <= len(similar_contacts_and_teams):
+                        selected_contact = similar_contacts_and_teams[int(choice) - 1]
+                        break
+
+            if selected_contact is not None:
+                self.associate_contact_to_location(selected_contact, location)
+
+    def associate_contact_to_location(self, contact, location):
+        role, status = self.prompt_for_role_and_status()
+
+        # If email or phone fields are present in the location but not the contact, update the contact fields
+        updated_fields = {}
+        if location.contact_phone and not contact.phone:
+            contact.phone = location.contact_phone
+            updated_fields["phone"] = location.contact_phone
+        if location.contact_email and not contact.email:
+            contact.email = location.contact_email
+            updated_fields["email"] = location.contact_email
+        if updated_fields:
+            try:
+                contact.validated_save()
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"Updated {contact._meta.model_name.title()} {contact.name} field(s): {updated_fields!r}"
+                    )
+                )
+            except ValidationError as e:
+                contact.refresh_from_db()
+                self.stdout.write(
+                    self.style.ERROR(f"Attempted to update {contact!r} field(s) but failed: {updated_fields!r}: {e}")
+                )
+
+        try:
+            contact_association = ContactAssociation(
+                contact=contact if isinstance(contact, Contact) else None,
+                team=contact if isinstance(contact, Team) else None,
+                associated_object=location,
+                role=role,
+                status=status,
+            )
+            contact_association.validated_save()
+            location.contact_name = ""
+            location.contact_email = ""
+            location.contact_phone = ""
+            location.validated_save()
+        except Exception as e:
+            self.stdout.write(self.style.ERROR(f"Failed to create association: {e}"))
+        else:
+            self.stdout.write(self.style.SUCCESS(f"Associated {contact!r} to location {location.display}"))
+
+    def prompt_for_role_and_status(self):
+        # Prompt for role
+        self.stdout.write("\nValid roles for this association:")
+        valid_roles = list(Role.objects.get_for_model(ContactAssociation))
```

Comment:
Probably should handle as an error case the situation where `valid_roles` or `valid_statuses` are empty? Maybe during initial startup of the command?